packages/flame/lib/src/components/spawn_component.dart

```diff
@@ -146,16 +184,17 @@ class SpawnComponent extends Component {
       period: _period,
       repeat: true,
       onTick: () {
-        final component = factory(amount);
+        final List<PositionComponent> components = multiFactory(amount);
         if (!selfPositioning) {
-          component.position = area!.randomPoint(
-            random: _random,
-            within: within,
-          );
+          components
+              .forEach((component) => component.position = area!.randomPoint(
+                    random: _random,
+                    within: within,
+                  ));
```

Comment:
Use a for-loop instead

Reply:
I noticed that intelliJ tries to lead me to for loops as well, why are these so bad?

Reply:
They are not that bad, but they are slightly less performant, and when not passing in an already existing function it doesn't give any extra readability.